### ISSUE TITLE
For metrics, when using `--auto-gen-config`, exclude files rather than raising `Max`

### ISSUE THREAD

**Issue by vergenzt** [bbatsov/rubocop#3387]
See [example gist](https://gist.github.com/vergenzt/2cb8b4ca96e719074728e57deccd91d1). At the moment, when you have a `Metrics/ClassLength` limit and you run `rubocop --auto-gen-config`, the generated `.rubocop_todo.yml` file takes the maximum class length of your existing files and sets that as the global max.

I'd strongly prefer if the generated config could simply exclude files that exceed the limit, allowing the limit to still be imposed on new files. (Or even better: if the limit for offending files could simply be raised locally! That way you could prevent the local problem from getting worse.)

---
## Expected behavior
1. Clone [gist](https://gist.github.com/vergenzt/2cb8b4ca96e719074728e57deccd91d1).
2. `rubocop --auto-gen-config`.
3. `.rubocop_todo.yml` contains:

``` yml
...
Metrics/ClassLength:
  Exclude:
    - 'example.rb'
```
## Actual behavior

`.rubocop_todo.yml` contains:

``` yml
...
Metrics/ClassLength:
  Max: 11
```
## RuboCop version

```
$ rubocop -V
0.40.0 (using Parser 2.3.1.0, running on ruby 2.1.0 x86_64-darwin14.0)
```

**Comment by Drenmi** [comment]
> allowing the limit to still be imposed on new files.

Oh, wow. I was starting to love this feature, and didn't even think about this, rather obvious, implication. 😥

**Comment by vergenzt** [comment]
Perhaps a syntax for setting a local limit could be:
```yaml
...
Metrics/ClassLength:
  LocalLimits:
    example.rb: 11
```

Looking through the code, it seems like something could probably be added to [`lib/rubocop/cop/mixin/code_length.rb`](https://github.com/bbatsov/rubocop/blob/429153801d1922bd7f2cc8ab577374cea7017630/lib/rubocop/cop/mixin/code_length.rb) to allow for per-file code-length limit exceptions (i.e. max class/module/method length).

**Comment by Drenmi** [comment]
Closing this as a duplicate of #3264.
